server/utils/cloudFunctions/functions/test/algoliaTests/algoliaFunctions.test.js

```diff
@@ -20,63 +20,66 @@ const { DEV_ADS_INDEX } =
-    const addedSnap = createRecordWrapper(snap);
+      const createRecordWrapper = 
+          firestoreWrap(devAlgoliaFunctions.devCreateRecord);
+      const addedSnap = createRecordWrapper(snap);
 
-    const expectedInput = {data: snap.data(), objectID: snap.id};
-    const expectedOutput = expectedInput;
-    expectedOutput.data.alteredByMockAlgolia = true;
-    assert.deepEqual(addedSnap, expectedOutput);
-    assert.isTrue(saveStub.calledOnce);
-    assert.isTrue(saveStub.calledWithMatch(expectedInput));
+      const expectedInput = {data: snap.data(), objectID: snap.id};
+      const expectedOutput = expectedInput;
+      expectedOutput.data.alteredByMockAlgolia = true;
+      assert.deepEqual(addedSnap, expectedOutput);
+      assert.isTrue(saveStub.calledOnce);
+      assert.isTrue(saveStub.calledWithMatch(expectedInput));
+    });
   });
-});
 
-describe("test_updateRecord", () => {
-  it('should call saveObject with correct values, return its output', () => {
-    const saveStub = 
-        sinon.stub(DEV_ADS_INDEX, "saveObject").callsFake(mockSaveObject);
-    const randomChange = firestoreMock.exampleDocumentSnapshotChange();
+  describe("test_updateRecord", () => {
+    it('should call saveObject with correct values, return its output', () => {
+      const saveStub = 
+          sinon.stub(DEV_ADS_INDEX, "saveObject").callsFake(mockSaveObject);
+      const randomChange = firestoreMock.exampleDocumentSnapshotChange();
 
-    const updateWrapper = 
-        firestoreWrap(devAlgoliaFunctions.devUpdateRecord);
-    const updatedChange = updateWrapper(randomChange);
+      const updateWrapper = 
+          firestoreWrap(devAlgoliaFunctions.devUpdateRecord);
+      const updatedChange = updateWrapper(randomChange);
 
-    const expectedInput = {data: randomChange.after.data(), 
-                           objectID: randomChange.after.id};
-    const expectedOutput = expectedInput;
-    expectedOutput.data.alteredByMockAlgolia = true;
-    assert.deepEqual(updatedChange, expectedOutput);
-    assert.isTrue(saveStub.calledOnce);
-    assert.isTrue(saveStub.calledWithMatch(expectedInput));
+      const expectedInput = {data: randomChange.after.data(), 
+                            objectID: randomChange.after.id};
+      const expectedOutput = expectedInput;
+      expectedOutput.data.alteredByMockAlgolia = true;
+      assert.deepEqual(updatedChange, expectedOutput);
+      assert.isTrue(saveStub.calledOnce);
+      assert.isTrue(saveStub.calledWithMatch(expectedInput));
+    });
   });
-});
 
-describe("test_deleteRecord", () => {
-  it('should call deleteObject with correct values, return its output', () => {
-    const deleteStub = 
-        sinon.stub(DEV_ADS_INDEX, "deleteObject").callsFake(mockDeleteObject);
-    const snap = firestoreMock.exampleDocumentSnapshot();
+  describe("test_deleteRecord", () => {
+    it('should call deleteObject with correct values, return its output', () => {
```

Comment:
Do you test for create, save, or deleteRecord getting called with incorrect values?

Reply:
Not currently. Since these functions are in practice firewall'd to only use ads created by your `adWriter.java`, I am presuming that the ads are well-formed/typed.